src/main/java/de/retest/recheck/TestCaseFinder.java

```diff
@@ -86,7 +88,41 @@ public static TestCaseFinder getInstance() {
 	 * @return The class name for the test case method in all stack traces.
 	 */
 	public Optional<String> findTestCaseClassNameInStack() {
-		return findTestCaseMethodInStack( toClassName() );
+		final Optional<String> testCaseClassName = findTestCaseMethodInStack( toClassName() );
+		if ( testCaseClassName.isPresent() ) {
+			return testCaseClassName;
+		} else {
+			return findTestCaseClassInStack();
+		}
+	}
+
+	/**
+	 * @return The class name for the test case method in all stack traces.
```

Comment:
I assume copy & paste error?

```suggestion
	 * @return The class name for the test being called from.
```

Reply:
copy & pase yes but thats the sentence, that's on almost every other method, including those who track for the class name, who already existed. So i kept it out of consistency. Mayle take a look at the other comments and tell me if they should be changed too or if i'm wrong.

Reply:
Yes, the other methods work on the basis of test case methods (respectively their annotations). But this method doesn't, so that's why I suggested the changes here and below.